File: desafios-python/desafio3/java/Main.java
// ============================================================
// Desafio 3: Validador de Senha
// Disciplina: Linguagem de Programação I - FATEC
// Descrição: Solicita uma senha ao usuário repetidamente
//            até que ela tenha pelo menos 8 caracteres.
// ============================================================

import java.util.Scanner;

public class Main {

    // ----------------------------
    // VERSÃO 2: Função Auxiliar
    // ----------------------------
    public static boolean validarSenha(String senha) {
        return senha.length() >= 8;
    }

    // ----------------------------
    // VERSÃO 1: Procedural
    // ----------------------------
    public static void versaoProcedural(Scanner sc) {
        System.out.println("========================================");
        System.out.println("  VERSAO 1 - Procedural");
        System.out.println("========================================");

        System.out.print("Crie uma senha (minimo 8 caracteres): ");
        String senha = sc.next();

        while (senha.length() < 8) {
            System.out.println("  X Senha muito curta (" + senha.length() + " caracteres). Tente novamente.");
            System.out.print("Crie uma senha (minimo 8 caracteres): ");
            senha = sc.next();
        }

        System.out.println("  OK Senha aceita! (" + senha.length() + " caracteres)");
    }

    // ----------------------------
    // VERSÃO 2: Modular com Funções
    // ----------------------------
    public static void versaoModular(Scanner sc) {
        System.out.println("========================================");
        System.out.println("  VERSAO 2 - Modular com Funcoes");
        System.out.println("========================================");

        System.out.print("Crie uma senha (minimo 8 caracteres): ");
        String senha = sc.next();

        while (!validarSenha(senha)) {
            System.out.println("  X Senha muito curta (" + senha.length() + " caracteres). Tente novamente.");
            System.out.print("Crie uma senha (minimo 8 caracteres): ");
            senha = sc.next();
        }

        System.out.println("  OK Senha aceita! (" + senha.length() + " caracteres)");
    }

    // ----------------------------
    // Programa Principal
    // ----------------------------
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("\n>>> DESAFIO 3: Validador de Senha <<<\n");
        System.out.println("Qual versao deseja executar?");
        System.out.println("  1 - Procedural");
        System.out.println("  2 - Modular com Funcoes");
        System.out.print("Escolha (1 ou 2): ");

        int escolha = sc.nextInt();

        if (escolha == 1)
            versaoProcedural(sc);
        else if (escolha == 2)
            versaoModular(sc);
        else {
            System.out.println("Opcao invalida. Executando versao modular por padrao.");
            versaoModular(sc);
        }

        sc.close();
    }
}

File: desafios-python/desafio3/cpp/main.cpp
// ============================================================
// Desafio 3: Validador de Senha
// Disciplina: Linguagem de Programação I - FATEC
// Descrição: Solicita uma senha ao usuário repetidamente
//            até que ela tenha pelo menos 8 caracteres.
// ============================================================

#include <iostream>
#include <string>
using namespace std;

// ----------------------------
// VERSÃO 2: Função Auxiliar
// ----------------------------
bool validarSenha(const string& senha) {
    return senha.length() >= 8;
}

// ----------------------------
// VERSÃO 1: Procedural
// ----------------------------
void versaoProcedural() {
    cout << "========================================" << endl;
    cout << "  VERSAO 1 - Procedural" << endl;
    cout << "========================================" << endl;

    string senha;
    cout << "Crie uma senha (minimo 8 caracteres): ";
    cin >> senha;

    while (senha.length() < 8) {
        cout << "  X Senha muito curta (" << senha.length() << " caracteres). Tente novamente." << endl;
        cout << "Crie uma senha (minimo 8 caracteres): ";
        cin >> senha;
    }

    cout << "  OK Senha aceita! (" << senha.length() << " caracteres)" << endl;
}

// ----------------------------
// VERSÃO 2: Modular com Funções
// ----------------------------
void versaoModular() {
    cout << "========================================" << endl;
    cout << "  VERSAO 2 - Modular com Funcoes" << endl;
    cout << "========================================" << endl;

    string senha;
    cout << "Crie uma senha (minimo 8 caracteres): ";
    cin >> senha;

    while (!validarSenha(senha)) {
        cout << "  X Senha muito curta (" << senha.length() << " caracteres). Tente novamente." << endl;
        cout << "Crie uma senha (minimo 8 caracteres): ";
        cin >> senha;
    }

    cout << "  OK Senha aceita! (" << senha.length() << " caracteres)" << endl;
}

// ----------------------------
// Programa Principal
// ----------------------------
int main() {
    cout << "\n>>> DESAFIO 3: Validador de Senha <<<\n" << endl;

    cout << "Qual versao deseja executar?" << endl;
    cout << "  1 - Procedural" << endl;
    cout << "  2 - Modular com Funcoes" << endl;
    cout << "Escolha (1 ou 2): ";

    int escolha;
    cin >> escolha;

    if (escolha == 1)
        versaoProcedural();
    else if (escolha == 2)
        versaoModular();
    else {
        cout << "Opcao invalida. Executando versao modular por padrao." << endl;
        versaoModular();
    }

    return 0;
}
